src/websockets.rs

```diff
@@ -109,57 +118,38 @@ impl<'a> WebSockets<'a> {
     pub fn disconnect(&mut self) -> Result<()> {
         if let Some(ref mut socket) = self.socket {
             socket.0.close(None)?;
-            Ok(())
-        } else {
-            bail!("Not able to close the connection");
+            return Ok(());
         }
+        bail!("Not able to close the connection");
```

Comment:
I don't get this part of the change, it looks like the new code will always hit this `bail!` statement.

Reply:
No it won't because the `return Ok(());` causes  a early return. 

https://medium.com/swlh/return-early-pattern-3d18a41bba8

Reply:
Oops, missed that line. Thanks!